OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyLogic.cs

```diff
@@ -741,31 +742,33 @@ bool SwitchTeamChat()
 		void UpdateCurrentMap()
 		{
 			var uid = orderManager.LobbyInfo.GlobalSettings.Map;
-			if (Map.Uid == uid)
+			if (MapPreview.Uid == uid)
 				return;
 
-			Map = Game.ModData.MapCache[uid];
-			if (Map.Status == MapStatus.Available)
+			MapPreview = Game.ModData.MapCache[uid];
+			if (MapPreview.Status == MapStatus.Available)
 			{
 				// Maps need to be validated and pre-loaded before they can be accessed
 				new Thread(_ =>
 				{
-					var map = Map;
-					map.CacheRules();
+					Map = new Map(MapPreview.Path);
+					Map.PreloadRules();
```

Comment:
Also note that the existing code was already doing this (CacheRules was a wrapper for PreloadRules).